core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/negotiation/ConsumerContractNegotiationManagerImpl.java

```diff
@@ -86,13 +89,17 @@ public void stop() {
     @Override
     public StatusResult<ContractNegotiation> initiate(ContractRequestMessage contractOffer) {
         var id = UUID.randomUUID().toString();
+        var callbackAddress = CallbackAddress.Builder.newInstance()
+                .uri(contractOffer.getCallbackAddress())
+                .events(Set.of("contract.negotiation.initiated"))
+                .build();
         var negotiation = ContractNegotiation.Builder.newInstance()
                 .id(id)
                 .correlationId(id)
                 .protocol(contractOffer.getProtocol())
                 .counterPartyId(contractOffer.getConnectorId())
-                .counterPartyAddress(contractOffer.getConnectorAddress())
-                .callbackAddresses(contractOffer.getCallbackAddress())
+                .counterPartyAddress(contractOffer.getCallbackAddress())
+                .callbackAddresses(List.of(callbackAddress))
```

Comment:
I think this `callbackAddresses` field shouldn't be filled here, because they are used for the "connector - client" interaction, not for the "connector - connector" one.

Reply:
I applied your feedback. However, this changes previous behavior as the `ContractNegotiationIntegrationTest` was explicitely checking for the callback addresses. Was this wrong before? @ndr-brt

Reply:
I'd say so, the `callbackAddresses` are set by the part that initiate the negotiation only in its side, because they don't make sense on the other side, so they shouldn't be passed over the protocol.